### ISSUE TITLE
--kv-store: failed to initialize bit sequence handler: Key not found in store

### ISSUE THREAD

**Issue by jgkamat** [docker/docker#14372]
Description of problem:
Since today or yesterday, I am unable to use --kv-store with consul when starting the daemon on the experimental branch. This breaks the overlay networking experimental for me.

`docker version`:
```
Client:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   71f8ca3
 Built:        Thu Jul  2 20:43:59 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   71f8ca3
 Built:        Thu Jul  2 20:43:59 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true
```
`docker info`:
```
Containers: 0
Images: 0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-21-generic
Operating System: Ubuntu 14.04 LTS
CPUs: 2
Total Memory: 3.66 GiB
Name: mechjeb.vpc.cloudera.com
ID: OE35:VSJZ:EU2H:OCS5:ZRY6:FIFT:KN2Y:J3WA:FMHT:7CA4:NZJ4:ZZ2D
WARNING: No swap limit support
Experimental: true
```

`uname -a`:
`Linux mechjeb.vpc.cloudera.com 3.19.0-21-generic #21-Ubuntu SMP Sun Jun 14 18:31:11 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`

Environment details (AWS, VirtualBox, physical, etc.):
AWS, Ubuntu 14.04 with vivid kernel.

How reproducible:
Always

Steps to Reproduce:
1. Start 14.04 machine, update kernel to 3.19
2. Start a consul server on localhost/remotely, as described [here](https://github.com/docker/docker/blob/master/experimental/compose_swarm_networking.md)

Actual Results:
Error, daemon dosent start

Expected Results:
Daemon starts.

Additional info:
I'm not sure what subcomponent this is a problem with, so i'm reporting it here. If this is a problem with libkv or libnetwork, i'll move it there if you ask :smile:.  I last tried this on tuesday, with the dev client, and everything worked then.

Daemon logs
```
root@mechjeb:~# docker -d --kv-store consul:172.16.3.211:8500
WARN[0000] Running experimental build                   
INFO[0000] Listening for HTTP on unix (/var/run/docker.sock) 
INFO[0000] [graphdriver] using prior storage driver "aufs" 
INFO[0000] Option DefaultDriver: bridge                 
INFO[0000] Option DefaultNetwork: bridge                
INFO[0000] Option OptionKVProvider: consul              
INFO[0000] Option OptionKVProviderURL: 172.16.3.211:8500 
INFO[0000] Firewalld running: false                     
FATA[0000] Error starting daemon: Error initializing network controller: error obtaining controller instance: failed to initialize vxlan id manager: failed to initialize bit sequence handler: Key not found in store
```

**Comment by mavenugo** [comment]
@mrjana @aboch seems like some recent change causing this issue.

**Comment by aboch** [comment]
It's a regression introduced in libnetwork which only affects the experimental multihost overlay driver.
It made it to docker/docker when we vendored the latest libnetwork code to bring in fixes for 1.7.1 release.
I am not aware of a work-around, working to the fix.

**Comment by jgkamat** [comment]
When I build docker on the master branch (just running `make binary`) the resulting binary dosen't seem to have all the libnetwork features (running with `--kv-store` results in `flag provided but not defined`). How do we build a docker binary that includes the libnetwork features from master?

**Comment by mrjana** [comment]
@jgkamat please build it as follows to build the experimental binary
```
DOCKER_EXPERIMENTAL=true sudo make
```

**Comment by vijayendrabvs** [comment]
Hi,

What is the ETA to resolve this? I'm not building/running libnetwork from src, but am following the instructions mentioned in the following links to setup libnetworking on Docker on a multi host setup -

https://github.com/docker/docker/tree/master/experimental
https://github.com/docker/libnetwork/blob/master/docs/overlay.md

But I run into the same vxlan ID manager initialization error at this step -

```
ubuntu@ip-10-0-1-190:~$ docker -d --kv-store=consul:localhost:8500 --label=com.docker.network.driver.overlay.bind_interface=eth0
WARN[0000] Running experimental build
FATA[0000] Error starting daemon: open /var/run/docker.pid: permission denied
ubuntu@ip-10-0-1-190:~$ sudo docker -d --kv-store=consul:localhost:8500 --label=com.docker.network.driver.overlay.bind_interface=eth0
WARN[0000] Running experimental build
WARN[0000] Udev sync is not supported. This will lead to unexpected behavior, data loss and errors. For more information, see https://docs.docker.com/reference/commandline/cli/#daemon-storage-driver-option
INFO[0000] Listening for HTTP on unix (/var/run/docker.sock)
INFO[0000] [graphdriver] using prior storage driver "devicemapper"
INFO[0000] Option DefaultDriver: bridge
INFO[0000] Option DefaultNetwork: bridge
INFO[0000] Option OptionKVProvider: consul
INFO[0000] Option OptionKVProviderURL: localhost:8500
INFO[0000] Firewalld running: false
FATA[0000] Error starting daemon: Error initializing network controller: error obtaining controller instance: failed to initialize vxlan id manager: failed to initialize bit sequence handler: Key not found in store
ubuntu@ip-10-0-1-190:~$
```

My docker version is shown below -

```
ubuntu@ip-10-0-1-190:~$ docker -v
Docker version 1.8.0-dev, build 2fe51ec, experimental
ubuntu@ip-10-0-1-190:~$
```

Also, is there a page that provides guidelines on how to run libnetwork from source?

Thanks,
Regards,
Vijay

**Comment by jgkamat** [comment]
@vijayendrabvs 
This is solved by pull #14402. You can test it out by checking c6dc6bcbb8d7e9caa0ef9e75e4d223cf310dad9f from that pull request and running
```
export DOCKER_EXPERIMENTAL=true
sudo make
```
as mrjana said above. This works fine for me. If you put these binaries in while your cluster is setting up and restarting the daemon, you can get the swarm experimental working again.
